mutex.go

```diff
@@ -44,10 +44,12 @@ var (
 // Lock waits up to duration d for l.Lock() to succeed.
 func Lock(l sync.Locker, d time.Duration) error {
 	done := make(chan struct{}, 1)
-	go func() {
+	ctx, cancel := context.WithTimeout(context.Background(), d*time.Second)
+	defer cancel()
+	go func(context.Context) {
 		l.Lock()
 		done <- struct{}{}
-	}()
+	}(ctx)
```

Comment:
No, in that example `slowOperation` is just a function where context happens to be the first argument, as is your func(context.Context) { ... }` - just a closure. It doesn't do anything with the context.

The context type is also nothing magical, it's just a type. More details are here: https://blog.golang.org/context.

What needs to happen is for the loop to take a context and exit on timeout here:
https://github.com/marcacohen/gcslock/blob/7fdf2a1c841994c956df3bc064da26b55f391501/mutex.go#L89-L98